src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionDispatcher.cs

```diff
@@ -256,6 +257,8 @@ public HttpConnectionDispatcher(HttpConnectionManager manager, ILoggerFactory lo
                         // Start the transport
                         connection.TransportTask = longPolling.ProcessRequestAsync(context, tokenSource.Token);
 
+                        connection.PreviousPollTask = currentRequestTcs.Task;
```

Comment:
We should move this outside the `if`, just in case a client re-polls during the first request. It would be rare because the first poll is "instant" but it could happen.

Reply:
Good catch.